lib/blocs/home_page_cubit.dart

```diff
@@ -93,181 +87,67 @@ class HomePageCubit extends Cubit<MapRouteState> {
+              state.toPlace,
+              correlationId,
             );
-          }
+      final Plan plan = await currentFetchPlanOperation.valueOrCancellation(
+        null,
+      );
+
+      if (plan == null) {
+        throw FetchCanceledByUserException("Cancelled by User");
+      } else if (plan.hasError) {
+        updateMapRouteState(state.copyWith(isFetching: false));
+        if (car) {
+          throw FetchOnlineCarException(plan.error.message);
         } else {
-          appReviewCubit.incrementReviewWorthyActions();
-          setPlan(plan);
+          throw FetchOnlinePlanException(plan.error.message);
         }
-      } on FetchOfflineRequestException catch (e) {
-        // TODO: Replace by proper error handling
-        // ignore: avoid_print
-        print("Failed to fetch plan: $e");
-        _showOnAndOfflineErrorAlert(
-            context,
-            "Offline mode is not implemented yet.",
-            false,
-            localization.commonError);
-      } on FetchOfflineResponseException catch (e) {
-        // TODO: Replace by proper error handling
-        // ignore: avoid_print
-        print("Failed to fetch plan: $e");
-        _showOnAndOfflineErrorAlert(
-            context,
-            "Offline mode is not implemented yet.",
-            false,
-            localization.commonError);
-      } on FetchOnlineRequestException catch (e) {
-        // TODO: Replace by proper error handling
-        // ignore: avoid_print
-        print("Failed to fetch plan: $e");
-        _showOnAndOfflineErrorAlert(context, localization.commonNoInternet,
-            true, localization.commonError);
-      } on FetchOnlineResponseException catch (e) {
-        // TODO: Replace by proper error handling
-        // ignore: avoid_print
-        print("Failed to fetch plan: $e");
-        _showOnAndOfflineErrorAlert(context, localization.searchFailLoadingPlan,
-            true, localization.commonError);
-      } catch (e) {
-        // TODO: Replace by proper error handling
-        // ignore: avoid_print
-        print("Failed to fetch plan: $e");
-        _showErrorAlert(context, e.toString());
-      }
-
-      try {
-        currentFetchAdOperation =
-            requestManagerCubit.fetchAd(context, state.toPlace);
-
-        final Ad ad = await currentFetchAdOperation.valueOrCancellation(null);
-        await updateMapRouteState(
-          state.copyWith(ad: ad),
-        );
-      } catch (e, stacktrace) {
-        await updateMapRouteState(
-          MapRouteState(
-              fromPlace: state.fromPlace,
-              toPlace: state.toPlace,
-              isFetching: state.isFetching,
-              showSuccessAnimation: state.showSuccessAnimation,
-              plan: state.plan),
-        );
-        // TODO: Replace by proper error handling
-        // ignore: avoid_print
-        print("Failed to fetch ad: $e");
-        // ignore: avoid_print
-        print(stacktrace);
+      } else {
```

Comment:
Could we turn this if / else around like a positive case first? 
if (allGood) setPlan
else if (somethingBad happened)

Reply:
Could we just get rid of this else part? I mean if it should never happen it most likely will never happen :)